src/java_bytecode/java_bytecode_language.cpp

```diff
@@ -256,6 +257,106 @@ static void infer_opaque_type_fields(
   }
 }
 
+/// Create if necessary and then return a reference to a symbol for a class
+/// literal (i.e. the result of a Classname.class expression in Java). A
+/// constant global symbol will be created if it doesn't already exist.
+/// \param class_ref: the raw bytecode expression for the class literal
+/// \param symbol_table: global symbol table; may gain a new symbol
+/// \return symbol expression for the Class literal.
+static symbol_exprt get_or_create_class_literal_symbol(
+  const exprt &class_ref,
+  symbol_tablet &symbol_table)
+{
+  const irep_idt &class_id = class_ref.type().get(ID_identifier);
+  symbol_typet java_lang_Class("java::java.lang.Class");
+  symbol_exprt symbol_expr(
+    id2string(class_id)+"@class_model",
+    java_lang_Class);
+
+  if(!symbol_table.has_symbol(symbol_expr.get_identifier()))
+  {
+    symbolt new_class_symbol;
+    new_class_symbol.name = symbol_expr.get_identifier();
+    new_class_symbol.type = symbol_expr.type();
+    // name without "java::" prefix:
```

Comment:
I think this comment should be removed.

Reply:
Not done; without it it is not clear what we're trying to clip by removing 6 chars.